### PR TITLE
[air/output] Fix trial status at end (more info + cut off)

### PR DESCRIPTION
**Pull request by krfricke**
## Why are these changes needed?

This PR ensures that the full trial status table is printed at the end of a Ray Tune run with the new output engine. Additionally, trial status data was previously always cut off - now we enforce that when `force= True`, all trial data is reported.

It also fixes a bug for showing the `more_info` field (how many more trials with a specific status are available).

Before:

```
╭──────────────────────────────────────────────────────────────────────────────────────────────────╮
│ Trial name                   status              loss     iter     total time (s)     iterations │
├──────────────────────────────────────────────────────────────────────────────────────────────────┤
│ easy_objective_d52e5_00000   TERMINATED    1.63294          20            2.12258             19 │
│ easy_objective_d52e5_00001   TERMINATED   -0.00503704       20            2.13938             19 │
│ easy_objective_d52e5_00002   TERMINATED    0.993741         20            2.12147             19 │
│ easy_objective_d52e5_00003   TERMINATED   -1.37468          20            2.12688             19 │
│ easy_objective_d52e5_00004   TERMINATED   -0.366993         20            2.12531             19 │
╰──────────────────────────────────────────────────────────────────────────────────────────────────╯
55 more TERMINATED
```

After
```
╭────────────────────────────────────────────────────────────────────────────────────────────────╮
│ Trial name                   status            loss     iter     total time (s)     iterations │
├────────────────────────────────────────────────────────────────────────────────────────────────┤
│ easy_objective_2535f_00000   TERMINATED   -8.7459         20            2.12678             19 │
│ easy_objective_2535f_00001   TERMINATED   -4.72291        20            2.13098             19 │
│ easy_objective_2535f_00002   TERMINATED    9.68591        20            2.12318             19 │
│ easy_objective_2535f_00003   TERMINATED    8.8669         20            2.13312             19 │
│ easy_objective_2535f_00004   TERMINATED   -7.76264        20            2.10994             19 │
│ easy_objective_2535f_00005   TERMINATED    9.7339         20            2.11871             19 │
│ easy_objective_2535f_00006   TERMINATED   -8.92678        20            2.12034             19 │
│ easy_objective_2535f_00007   TERMINATED   -4.14569        20            2.12235             19 │
│ easy_objective_2535f_00008   TERMINATED   -1.90578        20            2.12757             19 │
│ easy_objective_2535f_00009   TERMINATED   -9.29668        20            2.1048              19 │
│ easy_objective_2535f_00010   TERMINATED   -8.82601        20            2.11207             19 │
│ easy_objective_2535f_00011   TERMINATED    4.58302        20            2.11505             19 │
│ easy_objective_2535f_00012   TERMINATED   -3.19667        20            2.10649             19 │
│ easy_objective_2535f_00013   TERMINATED    0.276288       20            2.11077             19 │
│ easy_objective_2535f_00014   TERMINATED    3.16934        20            2.11101             19 │
│ easy_objective_2535f_00015   TERMINATED   -6.99527        20            2.11281             19 │
│ easy_objective_2535f_00016   TERMINATED    5.26072        20            2.10124             19 │
│ easy_objective_2535f_00017   TERMINATED   -1.17222        20            2.1232              19 │
│ easy_objective_2535f_00018   TERMINATED   -3.2927         20            2.103               19 │
│ easy_objective_2535f_00019   TERMINATED    8.52026        20            2.0961              19 │
│ easy_objective_2535f_00020   TERMINATED   -9.50308        20            2.11589             19 │
│ easy_objective_2535f_00021   TERMINATED    3.52898        20            2.10839             19 │
│ easy_objective_2535f_00022   TERMINATED    2.24067        20            2.10501             19 │
│ easy_objective_2535f_00023   TERMINATED   -6.59235        20            2.10195             19 │
│ easy_objective_2535f_00024   TERMINATED   -7.5678         20            2.11382             19 │
│ easy_objective_2535f_00025   TERMINATED    8.14493        20            2.10336             19 │
│ easy_objective_2535f_00026   TERMINATED    8.22056        20            2.1018              19 │
│ easy_objective_2535f_00027   TERMINATED   -7.76516        20            2.08713             19 │
│ easy_objective_2535f_00028   TERMINATED   -6.19463        20            2.10738             19 │
│ easy_objective_2535f_00029   TERMINATED   -2.16162        20            2.08866             19 │
│ easy_objective_2535f_00030   TERMINATED    6.58598        20            2.09207             19 │
│ easy_objective_2535f_00031   TERMINATED    6.69676        20            2.10172             19 │
│ easy_objective_2535f_00032   TERMINATED   -0.127672       20            2.10933             19 │
│ easy_objective_2535f_00033   TERMINATED   -9.29816        20            2.10509             19 │
│ easy_objective_2535f_00034   TERMINATED    7.68197        20            2.09605             19 │
│ easy_objective_2535f_00035   TERMINATED    6.29745        20            2.10823             19 │
│ easy_objective_2535f_00036   TERMINATED    8.52747        20            2.10674             19 │
│ easy_objective_2535f_00037   TERMINATED   -0.680396       20            2.09909             19 │
│ easy_objective_2535f_00038   TERMINATED    1.02905        20            2.11742             19 │
│ easy_objective_2535f_00039   TERMINATED   -7.29284        20            2.09985             19 │
│ easy_objective_2535f_00040   TERMINATED    0.253536       20            2.1162              19 │
│ easy_objective_2535f_00041   TERMINATED    6.71257        20            2.09051             19 │
│ easy_objective_2535f_00042   TERMINATED    3.99266        20            2.11268             19 │
│ easy_objective_2535f_00043   TERMINATED   -3.61558        20            2.1147              19 │
│ easy_objective_2535f_00044   TERMINATED    1.72299        20            2.09601             19 │
│ easy_objective_2535f_00045   TERMINATED   -5.96753        20            2.11121             19 │
│ easy_objective_2535f_00046   TERMINATED    7.91628        20            2.12614             19 │
│ easy_objective_2535f_00047   TERMINATED   -5.20284        20            2.10836             19 │
│ easy_objective_2535f_00048   TERMINATED    8.15891        20            2.10423             19 │
│ easy_objective_2535f_00049   TERMINATED    6.37054        20            2.1078              19 │
│ easy_objective_2535f_00050   TERMINATED   -8.98244        20            2.10171             19 │
│ easy_objective_2535f_00051   TERMINATED    8.82456        20            2.09652             19 │
│ easy_objective_2535f_00052   TERMINATED   -1.23827        20            2.10859             19 │
│ easy_objective_2535f_00053   TERMINATED   -6.53946        20            2.10545             19 │
│ easy_objective_2535f_00054   TERMINATED   -1.95374        20            2.10248             19 │
│ easy_objective_2535f_00055   TERMINATED   -1.79311        20            2.11117             19 │
│ easy_objective_2535f_00056   TERMINATED   -7.93258        20            2.10502             19 │
│ easy_objective_2535f_00057   TERMINATED   -1.69906        20            2.09228             19 │
│ easy_objective_2535f_00058   TERMINATED   -8.08596        20            2.1227              19 │
│ easy_objective_2535f_00059   TERMINATED    6.12904        20            2.09889             19 │
╰────────────────────────────────────────────────────────────────────────────────────────────────╯
```

## Checks

- [ ] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
    - [ ] I've added any new APIs to the API Reference. For example, if I added a 
           method in Tune, I've added it in `doc/source/tune/api/` under the 
           corresponding `.rst` file.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

**Comment by scottsun94** [comment]
I have two questions:

1. Why do values of `loss` and `total time` have different decimal places?

Screen Shot 2023-05-08 at 10 31 08 AM

2. is there a better sorting order for users?
  - shall we show the best-performing trials first?

**Review by gjoliver — APPROVED** [review]
thanks. just one nit.

File: python/ray/tune/experimental/output.py [hunk 1]
```diff
@@ -559,7 +562,9 @@ def _get_overall_trial_progress_str(self, trials):
         return f"Trial status: {result}"
 
     # TODO: Return a more structured type to share code with Jupyter flow.
-    def _get_heartbeat(self, trials, *sys_args) -> Tuple[List[str], _TrialTableData]:
+    def _get_heartbeat(
+        self, trials, *sys_args, force: bool = False
```
[the remark this answers is not in the document]
Comment (krfricke) [reply python/ray/tune/experimental/output.py:566]: Renamed!

**Comment by krfricke** [comment]
> I have two questions:
> 
> 1. Why do values of `loss` and `total time` have different decimal places?
> 
> Screen Shot 2023-05-08 at 10 31 08 AM
> 

They actually have the same decimal places, it's just that they are not zero-padded - (e.g. `2.103 == 2.10300`). This is a formatting preference. I personally prefer stripped strings (as we have) but also happy to add zero padding.

> 2. is there a better sorting order for users?
> 
> * shall we show the best-performing trials first?

Hm it's a good point. So far we've ordered them in order of processing, which also looks "nice" because the trial IDs are sorted. But we may want to make this configurable, especially if a `metric`/`mode` is given. 

I can do this in a follow-up

**Comment by scottsun94** [comment]
> One more question here is:
Error still opens up a modal/diaglog while Log opens the task detail page. The behavior is not consistent here

Ah. It makes sense.

> But we may want to make this configurable, especially if a metric/mode is given.

Yeah. I feel like users want to view the best-performing trials first. +1 to `especially if a metric/mode is given.`

**Pull request merged**